### DISCUSSION TITLE
How to test if hyprland is running

### DISCUSSION THREAD

**Discussion by swwind** [hyprwm/Hyprland#769] [Q&A] [answered]
Hello, I want to write a script to launch Hyprland after I login in tty, but I can not find a nice way to test whether if hyprland is running or not. And I end up writing this:

```bash
# ~/.zprofile
if [[ $(hyprctl monitors) == "HYPRLAND_INSTANCE_SIGNATURE was not set! (Is Hyprland running?)" ]]; then
  exec Hyprland
fi
```

This looks not very beautiful, so I would like to ask if there is a better way to achieve this.

P.S. this script works very well, I just think it is not so beautiful and elegant.

**Comment by vaxerski** [comment, marked as answer]
`printenv HYPRLAND_INSTANCE_SIGNATURE`

**Reply by swwind** [reply]
Thanks, but I found a new problem when I switch to another tty:

- I launched hyprland on tty2, everything works well.
- I switched to tty3, hyprland launched again. (`printenv HYPRLAND_INSTANCE_SIGNATURE` also fails)
- Then I have two independent hyprland sessions now.

Both my method and `printenv HYPRLAND_INSTANCE_SIGNATURE` does not check if there is another existing hyprland session on other TTYs.

**Reply by swwind** [reply]
Ok, I can check my tty number and make my script only works on tty1, and all things works now.
